rosgraph_monitor/include/rosgraph_monitor/monitor.hpp

```diff
@@ -136,6 +137,14 @@ class RosGraphMonitor
   /// @param statistics Incoming statistics list
   void on_topic_statistics(const rosgraph_monitor_msgs::msg::TopicStatistics & statistics);
 
+  /// @brief Generate a RosGraph message containing current graph state
+  /// @return A message containing current node information and timestamp
+  std::unique_ptr<rosgraph_monitor_msgs::msg::RosGraph> generate_rosgraph();
```

Comment:
It's a nitpick but can we avoid requiring heap allocation here? Maybe like:

```suggestion
  /// Clears existing state
  void fill_rosgraph_msg(rosgraph_monitor_msgs::msg::RosGraph & msg);
```

That way one could theoretically reuse the same message, or not, as the user chooses rather than the API dictates.

This could have the same end result for starters by looking in the node like:

```
auto msg = std::make_unique<rosgraph_monitor_msgs::msg::RosGraph>();
monitor_.fill_rosgraph_msg(*msg);
publisher_->publish(msg);
```

but leaves us a little more flexible